src/utils/RoomUpgrade.ts

```diff
@@ -22,15 +22,18 @@ import Modal from "../Modal";
 import { _t } from "../languageHandler";
 import ErrorDialog from "../components/views/dialogs/ErrorDialog";
 import SpaceStore from "../stores/SpaceStore";
+import Spinner from "../components/views/elements/Spinner";
 
 export async function upgradeRoom(
     room: Room,
     targetVersion: string,
     inviteUsers = false,
     handleError = true,
     updateSpaces = true,
+    awaitRoom = false,
 ): Promise<string> {
     const cli = room.client;
+    const modal = Modal.createDialog(Spinner, null, "mx_Dialog_spinner");
```

Comment:
could call this `spinnerModal` or something so the `modal.close()` 100 lines later is a bit clearer.